### DISCUSSION TITLE
Dynamically invoke a pipeline not present in the repositary

### DISCUSSION THREAD

**Discussion by vansh2552** [community/community#60215] [Actions]
### Select Topic Area

Show & Tell

### Body

I want to generate a workflow inside another workflow dynamically. For example parent.yaml a step should create a file named child.yaml, add some content to it and then execute it. But the issue is that to invoke a pipeline one has to create a yaml file inside the workflows folder in github repositary. I came across a solution that after creating the child.yaml I can commit the file inside my workflows folder but I don't want any changes in my repositary.  Is there any way to do so?

**Comment by nirav-gajera** [comment]
1. Create a workflow file,  call it parent.yaml, that will be triggered by a specific event or schedule. This workflow will be responsible for generating and executing the child workflow.

2. Inside the parent.yaml workflow, you can use a script or code step to generate the content of the child workflow dynamically.

3. Once you have generated the content of the child workflow, you can use the GitHub REST API to create a new workflow file in the .github/workflows directory of your repository. This can be done by making a PUT request to the GitHub API endpoint: PUT /repos/{owner}/{repo}/contents/.github/workflows/child.yaml.

4. After creating the child workflow file, you can trigger its execution using the Workflow Dispatch event. Use the GitHub REST API to send a POST request to the API endpoint: POST /repos/{owner}/{repo}/actions/workflows/{workflow_id}/dispatches. Provide the necessary parameters and payload in the request to trigger the child workflow.

**Reply by vansh2552** [reply]
Thanks for you reply. Can you give a simple implementation of this since I am not familiar with REST API.
